File: oaktrees/src/main/java/co/com/oaktrees/entity/Pedido.java
package co.com.oaktrees.entity;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
public class Pedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idPedido;
    private Date fecha;
    private float valorTotal;
    private int idTipoEntrega;
    private int idEstado;
    private String idUsuario;
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "carrito_pedido", joinColumns = @JoinColumn(name = "pedido_id"),
            inverseJoinColumns = @JoinColumn(name = "carrito_id"))
    private List<Producto> productos = new ArrayList<>();

    public Pedido(Date fecha, float valorTotal, int idTipoEntrega, int idEstado, String idUsuario) {
        this.fecha = fecha;
        this.valorTotal = valorTotal;
        this.idTipoEntrega = idTipoEntrega;
        this.idEstado = idEstado;
        this.idUsuario = idUsuario;
    }

    public Pedido() {
    }

    public int getIdPedido() {
        return idPedido;
    }

    public void setIdPedido(int idPedido) {
        this.idPedido = idPedido;
    }

    public Date getFecha() {
        return fecha;
    }

    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }

    public float getValorTotal() {
        return valorTotal;
    }

    public void setValorTotal(float valorTotal) {
        this.valorTotal = valorTotal;
    }

    public int getIdTipoEntrega() {
        return idTipoEntrega;
    }

    public void setIdTipoEntrega(int idTipoEntrega) {
        this.idTipoEntrega = idTipoEntrega;
    }

    public int getIdEstado() {
        return idEstado;
    }

    public void setIdEstado(int idEstado) {
        this.idEstado = idEstado;
    }

    public String getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(String idUsuario) {
        this.idUsuario = idUsuario;
    }

    public List<Producto> getProductos() {
        return productos;
    }

    public void setProductos(List<Producto> productos) {
        this.productos = productos;
    }
}

File: oakFront/src/app/models/pedido.ts
import { Producto } from "./producto";

export class Pedido {

    idPedido?: number;
    fecha: string;
    valorTotal: number;
    tipoEntrega: string;
    telefono: string;
    direccion: string;
    idEstado: string;
    idUsuario: string;
    comentario: string;
    productos: Array<Producto> = [];

    constructor(valorTotal: number, tipoEntrega: string, telefono: string, direccion:string, idEstado: string, 
        idUsuario: string, productos: Array<Producto>, comentario: string){
        this.valorTotal = valorTotal;
        this.tipoEntrega = tipoEntrega;
        this.telefono = telefono;
        this.direccion = direccion;
        this.idEstado = idEstado;
        this.idUsuario = idUsuario;
        this.productos = productos;
        this.comentario = comentario;
    }
}
